WL-3576 : Adding logic for email sender/recipient in Contact Us Tool

nicholaswilson100 opened the pull request.

nicholaswilson100 committed:
WL-3576 : Adding logic for email sender/recipient in Contact Us Tool

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -32,7 +33,12 @@
+    private SiteService siteService = null;
+
+
+	public void init(ServletConfig config) throws ServletException {
```
Indentation?

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -59,11 +68,17 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
+            }
+
+            Map<String, String> siteUpdaters = new HashMap<String, String>();
+            boolean hasViewPermission = securityService.unlock((userId!=null ? userId : ""), "roster.viewallmembers", site.getReference());
```
How can userId be null? as you're checking it at line 69.

But I don't think you need to pass in the userId at all do you as it should just use the current one for the request.
`org.sakaiproject.authz.api.SecurityService#unlock(java.lang.String, java.lang.String)`

nicholaswilson100 replied:
Doh! yes you're right...

buckett commented on src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java:
```diff
@@ -139,39 +129,11 @@ private String handleReport(final EntityView view, final Map<String, Object> par
-        } else {
-            senderAddress = sakaiProxy.getUser(userId).getEmail();
+        if (userId != null) {
+			senderAddress = sakaiProxy.getUser(userId).getEmail();
```
Indentation.

nicholaswilson100 committed:
WL-3576 : Indentation fix

nicholaswilson100 committed:
WL-3576 : Indentation fix

nicholaswilson100 committed:
WL-3576 : No need to pass in userId in securitycheck.

buckett commented on src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java:
```diff
@@ -189,9 +151,39 @@ private String handleReport(final EntityView view, final Map<String, Object> par
                 }
             } else {
                 // The site has a contact email. Use it as the toAddress.
```
The comment isn't correct.

nicholaswilson100 replied:
Removed comment

buckett commented on src/java/org/sakaiproject/feedback/util/SakaiProxy.java:
```diff
@@ -259,10 +255,8 @@ public void sendEmail(String fromUserId, String senderAddress, String toAddress,
-            // Copy the sender in
-			msg.addRecipient(RecipientType.CC, fromName, fromAddress);
-		}
+        // Copy the sender in
```
Indentation.

nicholaswilson100 replied:
Fixed below.

buckett commented on src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java:
```diff
@@ -189,9 +151,39 @@ private String handleReport(final EntityView view, final Map<String, Object> par
-        }
+
+		}
+		else {
```
Code style change

nicholaswilson100 replied:
Got it

buckett commented:
FeedbackEntityProvider.java#141 Still references the old logic.

buckett commented on src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:
```diff
@@ -59,11 +68,17 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
 
         if (userId != null) {
```
According to the logic in the Jira we weren't going todo checks to see if the user was logged in.

nicholaswilson100 replied:
I've moved the permission stuff out of the userId==null check.

buckett commented on src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java:
```diff
@@ -189,9 +151,39 @@ private String handleReport(final EntityView view, final Map<String, Object> par
             } else {
                 // The site has a contact email. Use it as the toAddress.
-                toAddress = sakaiProxy.getSiteProperty(siteId, Site.PROP_SITE_CONTACT_EMAIL);
+                toAddress = sakaiProxy.getConfigString(Constants.PROP_TECHNICAL_ADDRESS, null);
```
If the user is logged in and doesn't have permission to see the site members then the email will always go to the central technical address and never the site contact email.

nicholaswilson100 replied:
Good spot. Added logic to say that the email for a logged in user with no permission is the site contact email or the technical one if that doesn't exist.

nicholaswilson100 committed:
WL-3576 : Fix indentation (for my line and a few other lines around it)

nicholaswilson100 committed:
WL-3576 : Fix code style change

nicholaswilson100 committed:
WL-3576 : Acc to Jira we weren't going to do checks if the user was logged in

I've moved the security stuff out of the userId==null check.
But we still need to/can do this check as we put different attributes in the request depending on whether the user is logged in or not

nicholaswilson100 committed:
WL-3576 : Remove irrelevant comment

nicholaswilson100 committed:
WL-3576 : Fix loggedin, no perms user will always go to technical email

If the user is logged in and doesn't have permission to see the site members then the email will always go to the central technical address and never the site contact email.

buckett commented:
Won't merge cleanly.

nicholaswilson100 commented:
Hmmm... should we merge it un-cleanly then I can fix it up after?

nicholaswilson100 committed:
Merge remote-tracking branch 'origin/weblearn-10.x' into WL-3576

* origin/weblearn-10.x:
  WL-3566 : Add if-else logic to which email recipient to show
  WL-3566 : move click attaching to on page load rather than every method call
  WL-3566 : Should be doing the fit frame after changing the html
  WL-3592 : contact list is empty
  WL-3566 : Indentation
  WL-3567 : Use dbWrite method as we don't care about return value
  WL-3576 : Add email sent confirmation message to feedback tool
  WL-3567 : Add autoincrement column to avoid db insert exeption

Conflicts:
	src/java/org/sakaiproject/feedback/tool/FeedbackTool.java
	src/webapp/js/feedback.js
